### PR TITLE
[16.0][MIG] crm_location

### PR DESCRIPTION
**Pull request by jesusVMayor**

### CODE REVIEW

File: crm_location/models/crm_lead.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            self.update(
+                {
+                    "zip": self.location_id.name,
+                    "city": self.location_id.city_id.name,
+                    "state_id": self.location_id.city_id.state_id,
+                    "country_id": self.location_id.city_id.country_id,
+                }
+            )
+
+    location_id = fields.Many2one(
```
Comment (rousseldenis) [inline crm_location/models/crm_lead.py]: Could you put fields on top ?

File: crm_location/models/crm_lead.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+from odoo import api, fields, models
+
+
+class CrmLead(models.Model):
+    _inherit = "crm.lead"
+
+    @api.onchange("location_id")
+    def on_change_city(self):
+        if self.location_id:
```
Comment (rousseldenis) [inline crm_location/models/crm_lead.py:25]: As fields have been changed to computed readonly False, could you override _compute_partner_address_values method instead?

File: crm_location/models/crm_lead.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    location_id = fields.Many2one(
+        comodel_name="res.city.zip",
+        string="Location",
+        index="btree",
+        help="Use the city name or the zip code to search the location",
+    )
+
+    @api.onchange("partner_id")
+    def onchange_partner_id_crm_location(self):
```
Comment (rousseldenis) [inline crm_location/models/crm_lead.py]: You can change location_id as computed readonly False field too

File: crm_location/tests/test_crm_location.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+# Copyright 2019 Tecnativa - Alexandre Díaz
+# License AGPL-3 - See https://www.gnu.org/licenses/agpl-3.0
+
+from odoo.tests import common
+
+
+class TestCrmLocation(common.TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestCrmLocation, cls).setUpClass()
```
Comment (rousseldenis) [inline crm_location/tests/test_crm_location.py]:
```suggestion
        super().setUpClass()
```

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review.

@jesusVMayor Could you also change pre-commit message to refer good module name ?

image

**Comment by jesusVMayor** [comment]
@rousseldenis i made the changes, thanks for the review.

**Review by rousseldenis — APPROVED** [review]
Code review

**Comment by rafaelbn** [comment]
@moduon MT-1664

**Review by flotho — APPROVED** [review]
tested on runboat,
works as expected
LGTM

**Pull request merged**